### PR TITLE
Synchronise documentations with the project file structure.

### PR DESCRIPTION
**Pull request by ithinuel**
Some files haves been renamed/moved in the project while their reference in comments & documentation have not been updated.

This fixes (at least some of) those discrepancies.

### CODE REVIEW

File: doc/develop/optimizations/tools.rst [hunk 1]
```diff
@@ -229,32 +229,32 @@ as in the following example.
 
 Pahole will generate something similar to the output below in the console::
 
-    /* Used at: zephyr/isr_tables.c */
-    /* <80> ../include/sw_isr_table.h:30 */
-    struct _isr_table_entry {
-            void *                     arg;                  /*     0     4 */
-            void                       (*isr)(void *);       /*     4     4 */
+    /* Used at: […]/build/zephyr/kobject_hash.c */
+    /* <375> […]zephyr/include/zephyr/sys/dlist.h:37 */
```
Comment (tomi-font) [inline doc/develop/optimizations/tools.rst:233]: ditto

File: doc/develop/optimizations/tools.rst [hunk 2]
```diff
@@ -229,32 +229,32 @@ as in the following example.
...
-    /* <112> ../include/arch/arm/aarch32/cortex_m/mpu/arm_mpu.h:24 */
-    struct arm_mpu_region {
-            uint32_t                   base;                 /*     0     4 */
-            const char  *              name;                 /*     4     4 */
-            arm_mpu_region_attr_t      attr;                 /*     8     4 */
-
-            /* size: 12, cachelines: 1, members: 3 */
-            /* last cacheline: 12 bytes */
+    /* Used at: […]/build/zephyr/kobject_hash.c */
+    /* <3b7> […]zephyr/include/zephyr/sys/dlist.h:41 */
```
Comment (tomi-font) [inline doc/develop/optimizations/tools.rst:254]:
```suggestion
    /* <3b7> […]/zephyr/include/zephyr/sys/dlist.h:41 */
```
not consistent

**Review by tomi-font — DISMISSED** [review]

File: doc/develop/optimizations/tools.rst (hunk 2 above)
Comment (ithinuel) [reply doc/develop/optimizations/tools.rst]: Good catch thank you !

File: doc/develop/optimizations/tools.rst [hunk 3]
```diff
@@ -229,32 +229,32 @@ as in the following example.
 
 Pahole will generate something similar to the output below in the console::
 
-    /* Used at: zephyr/isr_tables.c */
-    /* <80> ../include/sw_isr_table.h:30 */
-    struct _isr_table_entry {
-            void *                     arg;                  /*     0     4 */
-            void                       (*isr)(void *);       /*     4     4 */
+    /* Used at: […]/build/zephyr/kobject_hash.c */
```
Comment (kartben) [inline doc/develop/optimizations/tools.rst:232]: I don't think we've quite settled on what to do with Unicode characters so in the meanwhile do you mind doing `s/…/.../g`?

**Review by kartben — CHANGES_REQUESTED** [review]

**Review by kartben — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**